src/java.base/share/classes/java/lang/invoke/LambdaMetafactory.java

```diff
@@ -242,6 +229,20 @@
  * used, but this is not compatible with some implementation techniques and
  * would complicate the work implementations must do.
  *
+ * <p>Uses besides evaluation of lambda expressions and method references are
+ * unintended.  These linkage methods may change their unspecified behaviors at
+ * any time to better suit the Java language features they were designed to
+ * support, and such changes may impact unintended uses.  Unintended uses of
+ * these linkage methods may lead to resource leaks, or other unspecified
+ * negative effects.
+ *
+ * @implNote In the reference implementation, the classes implementing the created
+ * function objects are strongly reachable from the defining class loader of the
+ * caller.  This technique reduces heap memory use, but it prevents the
+ * implementation class from unloading when the caller is a {@linkplain
+ * MethodHandles.Lookup.ClassOption#STRONG hidden class} and is unloaded, or when
+ * the returned call site is no longer strongly reachable.
```

Comment:
I don't get the last part of the sentence here. When the call site is no longer reachable _that_ prevents the class from being unloaded?

I like the implNote though. Maybe: "This technique reduces heap memory use, but it prevents the implementation class from being unloaded independently of its defining class loader" ?

Reply:
Re this sentence: I wish to indicate "caller is unloaded and impl class is unreachable elsewhere (besides in the ClassLoader)".  Maybe just "caller is unloaded" is fine. Thoughts on phrasing?

Reply:
Maybe 'caller's class is unloaded'.

Also, is the link correct? It links to the STRONG option, but the link text is 'hidden class'.

Reply:
The link is intended: I wish the link to describe the unloadable scenario for hidden classes.